### PR TITLE
build: preserve correct mode when generating files via jinja2

### PR DESCRIPTION
**Pull request by brauner**
When using "capture : true" in custom_target()s the mode of the source
file is not preserved when the generated file is not installed and so
needs to be tweaked manually. Switch from output capture to creating the
target file and copy the permissions from the input file.

### CODE REVIEW

File: units/meson.build [hunk 1]
```diff
@@ -272,8 +272,7 @@ foreach tuple : in_units
                 file,
                 input : file + '.in',
                 output : file,
-                command : [meson_render_jinja2, config_h, '@INPUT@'],
-                capture : true,
+		command : [meson_render_jinja2, config_h, '@INPUT@', '@OUTPUT@'],
```
Comment (bluca) [inline units/meson.build]: we use spaces-only for indentation (applies to all meson files)

**Review by bluca — APPROVED** [review]
lgtm, but tabs need to be converted to spaces

File: units/meson.build (hunk 1 above)
Comment (brauner) [reply units/meson.build]: Bah, I forgot about that.

Comment (brauner) [reply units/meson.build]: Done.

**Comment by keszybz** [comment]
LGTM, thanks!

**Pull request merged**